src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
We introduced snapshot mostly to target configuration reload, originally when we introduced this in 1.1, this was simply capturing the current OptionsMonitor.Value into a scoped service.  In 2.0 we were removed the caching so it would be recomputed every request (It was exactly the same as IOptions just with a scoped lifetime).  

Using the new stuff, I'm guessing if you wire up a scoped cache for TOptions, that will have the same behavior as the 2.0 IOptionsSnapshot, where it will get recomputed every request.  To get the more advanced cache invalidation behavior we had in 1.1, we could wire up the change token for configurationReload to explicitly evict the entry for TOptions.  I think the latter scenario is the important one, the change we made for snapshot for 2.0 was looking at supporting snapshots that used DbContext's to configure options, which will now also be able to manipulate the cache directly to populate options (so I think its also covered)